shepherd/shepherd/shepherd.py

```diff
@@ -64,9 +65,13 @@ def __init__(self,
             self.sheep[sheep_id] = sheep
             self.poller.register(socket, zmq.POLLIN)
             gevent.spawn(partial(self.dequeue_and_feed_jobs, sheep_id))
-            gevent.spawn(partial(self.health_check, sheep_id))
+            gevent.spawn(partial(self.sheep_health_check, sheep_id))
```

Comment:
Should not we store the greenlets and end them gracefully if need be? (e.g.: as we do on line 71)